### PR TITLE
volatility: 2.6.1 -> unstable-2020-12-12, fix the build

### PR DESCRIPTION
**Pull request by emilytrau**
###### Description of changes

Volatility is a memory forensics tool. Although it has been superseded by volatility3, volatility2 still sees regular use. As python2 support is dropping from its dependencies, this will need further future maintenance  as long as the tool is yet to be obsoleted. 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/volatility/default.nix [hunk 1]
```diff
@@ -2,23 +2,31 @@
...
   src = fetchFromGitHub {
     owner = "volatilityfoundation";
     repo = pname;
-    rev = version;
-    sha256 = "1v92allp3cv3akk71kljcwxr27h1k067dsq7j9h8jnlwk9jxh6rf";
+    rev = "a438e768194a9e05eb4d9ee9338b881c0fa25937";
+    sha256 = "U/iWPztn47/Tcz/etVduG7V/Nu5Z1jb+i0mutaKr9sM=";
   };
 
   doCheck = false;
```
Comment (pbsds) [inline pkgs/tools/security/volatility/default.nix:14]:
```suggestion
  doCheck = false;
  pythonImportsCheck = [ "volatility" ];
```

File: pkgs/development/python2-modules/openpyxl/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  postPatch = ''
+    # LICENSE.rst is missing, and setup.cfg currently doesn't contain anything useful anyway
+    # This should likely be removed in the next update
+    rm setup.cfg
+  '';
+
+  # Tests are not included in archive.
+  # https://bitbucket.org/openpyxl/openpyxl/issues/610
+  doCheck = false;
```
Comment (pbsds) [inline pkgs/development/python2-modules/openpyxl/default.nix:30]:
```suggestion
  doCheck = false;
  pythonImportsCheck = [ "openpyxl" ];
```

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 184782` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>volatility</li>
  </ul>
</details>

**Review by pbsds — CHANGES_REQUESTED** [review]

**Review by pbsds — APPROVED** [review]

**Comment by JJJollyjim** [comment]
+1, volatility2 is still useful, tested this PR and it works well.

**Comment by erdnaxe** [comment]
Result of `nixpkgs-review pr 184782` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>volatility</li>
  </ul>
</details>

---

Volatility2 is still useful for forensic security. IIRC Volatility3 has a different feature set.

File: pkgs/development/python2-modules/openpyxl/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+buildPythonPackage rec {
+  pname = "openpyxl";
+  version = "2.6.4";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1qzjj8nwj4dn0mhq1j64f136afiqqb81lvqiikipz3g1g0b80lqx";
+  };
+
+  checkInputs = [ pytest ];
```
Comment (SuperSandro2000) [inline pkgs/development/python2-modules/openpyxl/default.nix:19]: Please use pytestCheckHook

File: pkgs/development/python2-modules/openpyxl/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  postPatch = ''
+    # LICENSE.rst is missing, and setup.cfg currently doesn't contain anything useful anyway
+    # This should likely be removed in the next update
+    rm setup.cfg
+  '';
+
+  # Tests are not included in archive.
+  # https://bitbucket.org/openpyxl/openpyxl/issues/610
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python2-modules/openpyxl/default.nix:30]: Why not download from bitbucket instead?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
python2 is EOL. Has the tool been ported to python3?

**Comment by JJJollyjim** [comment]
No, Volatility only supports Python 2. Volatility3 was meant to be the python3 port but it is less useful and less popular.

**Comment by mweinelt** [comment]
We are well past the time, where we support adding new python2 packages. In fact at this point we only support python2 as a build time dependency, not as a runtime dependency.

**Pull request closed without merging**